service/cluster/xtra_db_cluster.go

```diff
@@ -50,6 +50,58 @@ func NewXtraDBClusterService(p *message.Printer) *XtraDBClusterService {
 	return &XtraDBClusterService{p: p}
 }
 
+// setProxySQLParams sets proxy's compute resources and disk size to given params.
+// The function was created to increase readablility.
+func setProxySQLParams(proxy *k8sclient.ProxySQL, params *controllerv1beta1.XtraDBClusterParams) error {
+	proxySQLDiskSize, err := convertors.StrToBytes(proxy.DiskSize)
+	if err != nil {
+		return err
+	}
+	params.Proxysql = &controllerv1beta1.XtraDBClusterParams_ProxySQL{
+		DiskSize: int64(proxySQLDiskSize),
+	}
+
+	if proxy.ComputeResources == nil {
+		return nil
+	}
+	cpuMillis, err := convertors.StrToMilliCPU(proxy.ComputeResources.CPUM)
+	if err != nil {
+		return err
+	}
+	memoryBytes, err := convertors.StrToBytes(proxy.ComputeResources.MemoryBytes)
+	if err != nil {
+		return err
+	}
+	params.Proxysql.ComputeResources = &controllerv1beta1.ComputeResources{
+		CpuM:        int32(cpuMillis),
+		MemoryBytes: int64(memoryBytes),
+	}
```

Comment:
looks like it can be extracted and reused

Reply:
Done